spotbugs/src/main/java/edu/umd/cs/findbugs/detect/InefficientReplaceAll.java

```diff
@@ -0,0 +1,143 @@
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;"));
+
+    private final BugAccumulator bugAccumulator;
+    
+    private static List<String> regexCharList = new ArrayList<>();
+    static {
+        regexCharList.add(".");
+        regexCharList.add("\\");
+        regexCharList.add("[");
+        regexCharList.add("]");
+        regexCharList.add("{");
+        regexCharList.add("}");
+        regexCharList.add("(");
+        regexCharList.add(")");
+        regexCharList.add("<");
+        regexCharList.add(">");
+        regexCharList.add("*");
+        regexCharList.add("+");
+        regexCharList.add("-");
+        regexCharList.add("=");
+        regexCharList.add("?");
+        regexCharList.add("^");
+        regexCharList.add("$");
+        regexCharList.add("|");
+    }
+
+    public InefficientReplaceAll(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (hasInterestingMethod(classContext.getJavaClass().getConstantPool(), methods)) {
+            classContext.getJavaClass().accept(this);
+        }
+    }
+
+    @Override
+    public void visit(Method obj) {
+        if (DEBUG) {
+            System.out.println("------------------- Analyzing " + obj.getName() + " ----------------");
+        }
+        super.visit(obj);
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (DEBUG) {
+            System.out.println("Opcode: " + Const.getOpcodeName(seen));
+        }
+        if (((seen == Const.INVOKEVIRTUAL) || (seen == Const.INVOKEINTERFACE)) && ("replaceAll".equals(getNameConstantOperand()))
+                && ("(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;".equals(getSigConstantOperand()))
+                && hasConstantArguments()) {
+            String firstArgument = getFirstArgument();
+            boolean found = false;
+            for (String regexChar : regexCharList) {
+                if (firstArgument.contains(String.valueOf(regexChar))) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                bugAccumulator.accumulateBug(
+                        new BugInstance(this, "IRA_INEFFICIENT_REPLACEALL", LOW_PRIORITY).addClassAndMethod(this), this);                    
+            }
+        }
+    }
+    
+    /**
+     * @return first argument of the called method if it's a constant
+     */
+    private String getFirstArgument() {
+        Object value = getStack().getStackItem(getNumberArguments(getMethodDescriptorOperand().getSignature()) - 1)
+                .getConstant();
+        return value == null ? null : value.toString();
```

Comment:
This could be simplified with `Objects.toString` and perhaps returning an `Optional`.